packages/outstatic/src/utils/yup.ts

```diff
@@ -27,4 +27,20 @@ export const editDocumentSchema: yup.SchemaOf<Document> = yup.object().shape({
     .required(),
   description: yup.string(),
   coverImage: yup.string()
-})
+}
+
+export const editDocumentSchema: yup.SchemaOf<SchemaShape> = yup
+  .object()
+  .shape(documentShape)
+
+export const convertSchemaToYup = (customFields: CustomFields) => {
```

Comment:
This is heavily coupled to the schema definition. Maybe consider defining the custom fields using json-schema and then just using a yup transformer [1](https://www.npmjs.com/package/json-schema-to-yup), [2](https://www.npmjs.com/package/json-schema-yup-transformer)